src/security_buffer.rs

```diff
@@ -26,82 +19,224 @@ pub enum SecurityBuffer<'data> {
     Empty,
 }
 
+/// A special security buffer type is used for the data decryption. Basically, it's almost the same
+/// as `OwnedSecurityBuffer` but for decryption.
```

Comment:
thought: What about `SecurityBuffer` for the owned version, and `SecurityBufferRef` for the borrowing version? It’s a common naming I’m seeing more and more. (Follow up in a separate PR.)

Reply:
Good idea! Your naming suggestions are good as always. Somewhat I didn't think about the `-Ref` postfix.
Done